### PR TITLE
feat: add sample code for using regional Dialogflow endpoint

### PR DESCRIPTION
**Pull request by martini9393**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-dialogflow/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> 🦕

### CODE REVIEW

File: samples/snippets/detect_intent_texts_with_location.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python
+
+# Copyright 2017 Google LLC
```
Comment (crwilcox) [inline samples/snippets/detect_intent_texts_with_location.py:3]: is this copied or new, if so this should be rev'd

File: samples/snippets/detect_intent_texts_with_location_test.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+from __future__ import absolute_import
+
+import os
+import uuid
+
+from detect_intent_texts_with_location import detect_intent_texts_with_location
+
+PROJECT_ID = os.getenv('GOOGLE_CLOUD_PROJECT')
+LOCATION_ID = 'europe-west2'
```
Comment (crwilcox) [inline samples/snippets/detect_intent_texts_with_location_test.py:22]: Definitely a nit, should this also be a variable? Also seeing eu-west regions with `en-us` encoding feels odd?

File: samples/snippets/detect_intent_texts_with_location.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  python detect_intent_texts_with_location.py --project-id PROJECT_ID \
+  --location-id LOCATION_ID --session-id SESSION_ID \
+  "hello" "book a meeting room" "Mountain View"
+"""
+
+import argparse
+import uuid
+
+
+# [START dialogflow_detect_intent_text_with_location]
```
Comment (crwilcox) [inline samples/snippets/detect_intent_texts_with_location.py:30]: Putting the region tag outside of the method will make the sample render with the method name/def. Is that intended? 

For instance https://github.com/googleapis/python-bigquery/blob/master/samples/snippets/authenticate_service_account.py shows how you might just want the content of the method, not the method itself.

File: samples/snippets/detect_intent_texts_with_location.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+def detect_intent_texts_with_location(project_id, location_id, session_id, texts, language_code):
+    """Returns the result of detect intent with texts as inputs.
+
+    Using the same `session_id` between requests allows continuation
+    of the conversation."""
+    from google.cloud import dialogflow
+    session_client = dialogflow.SessionsClient(client_options={
+       "api_endpoint": "{location}-dialogflow.googleapis.com".format(location=location_id)})
+
+    session = "projects/{project}/locations/{location}/agent/sessions/{session}".format(
```
Comment (crwilcox) [inline samples/snippets/detect_intent_texts_with_location.py:40]: a bit of a syle choice, but I think you can make all of your format strs into f-strs if you'd like, as our minimum supported version is Python 3.6.

`session = f"projects/{project_id}/locations/{location_id}/agent/sessions/{session_id}"`

**Review by crwilcox — APPROVED** [review]

File: samples/snippets/detect_intent_texts_with_location.py (hunk 1 above)
Comment (martini9393) [reply samples/snippets/detect_intent_texts_with_location.py:3]: updated to 2021.

File: samples/snippets/detect_intent_texts_with_location.py (hunk 4 above)
Comment (martini9393) [reply samples/snippets/detect_intent_texts_with_location.py:40]: done

File: samples/snippets/detect_intent_texts_with_location.py (hunk 3 above)
Comment (martini9393) [reply samples/snippets/detect_intent_texts_with_location.py:30]: I think that's intended, I followed the current style of other samples.

File: samples/snippets/detect_intent_texts_with_location_test.py (hunk 2 above)
Comment (martini9393) [reply samples/snippets/detect_intent_texts_with_location_test.py:22]: do you mean making it an environmental variable? I would prefer to also show users the format of location, as some people also get confused about that.

I updated en-US to en-GB, as europe-west2 is in London.

**Pull request merged**